source/extensions/filters/http/common/stream_rate_limiter.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/event/timer.h"
+#include "envoy/runtime/runtime.h"
+
+#include "common/buffer/watermark_buffer.h"
+#include "common/common/token_bucket_impl.h"
+
+namespace Envoy {
+
+class ScopeTrackedObject;
+
+namespace Event {
+class Timer;
+} // namespace Event
+
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+
+/**
+ * An HTTP stream rate limiter. Used in the fault filter and bandwidth filter.
```

Comment:
If it's going to be in the common folder, this should comment be a bit beefier. Why would other filter's use this and what does it do for you, etc.

Reply:
Sure